### PR TITLE
Allow users to specify the Azure RG to deploy into

### PR DESCRIPTION
**Pull request by iameskild**
## Reference Issues or PRs

Allow users to specify (via the nebari-config.yaml) which Azure Resource Group they deploy into. This also tries to preserve backward-compatibility so if nothing is provided, the default (current) behavior will be the same.

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

**Comment by iameskild** [comment]
I would still like to add a resource group name validator.

**Comment by iameskild** [comment]
The kubernetes test will fail until #1856 is merged.

File: src/_nebari/utils.py
Comment (Adam-D-Lewis) [inline src/_nebari/utils.py:289]: It seems a little confusing to have a resource_group_name passed in which is not what is returned by the set_azure_resource_group_name.  My preference would be to call the arg `base_resource_group_name`.

File: src/_nebari/utils.py
Comment (Adam-D-Lewis) [inline src/_nebari/utils.py:286]: I don't love the set_azure_resource_group_name since we aren't setting anything.  I'd just call it something along the lines of `azure_resource_group_name`, `build_azure_resource_group_name`, `construct_resource_group_name`, etc.

**Comment by Adam-D-Lewis** [comment]
Minor comments.  LGTM otherwise assuming it runs.  I'll approve.

**Review by Adam-D-Lewis — APPROVED** [review]

**Comment by iameskild** [comment]
@Adam-D-Lewis I tested a few permutations of this (alongside some of the changes in #1930) and things appear to be working well. I'd like to add a test for this in our integration tests.

**Pull request merged**
